docs/index.rst

```diff
@@ -0,0 +1,24 @@
+##########################
+edx-platform Documentation
+##########################
+
+Developer documentation is split between this docs directory and documentation
```

Comment:
I simplified this and try to let docs/index.rst do the explaining to avoid duplication.